packages/cid-verifier/src/verification.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-env serviceworker, browser */
+/* global Response */
+import pRetry from 'p-retry'
+import { getFromDenyList } from './utils/denylist'
+
+/**
+ * Get the google malware result stored in KV.
+ *
+ * @param {string} cid
+ * @param {import('./env').Env} env
+ */
+async function getGoogleMalwareResult (cid, env) {
+  const datastore = env.GOOGLE_EVALUATE_RESULTS
+  if (!datastore) {
+    throw new Error('db not ready')
+  }
+
+  return await pRetry(
+    () => datastore.get(cid),
+    { retries: 5 }
+  )
```

Comment:
This should do a KV list operation with cid prefix to see if we have any result: either .lock , or a `cid/service` kind of key with results already available